noir-projects/noir-protocol-circuits/crates/rollup-lib/src/components.nr

```diff
@@ -349,126 +361,99 @@ fn get_tx_effects_hash_input(
     }
 
     // Now we know the number of fields appended, we can assign the first value:
-    // TX_START_PREFIX | 0 | txlen[0] txlen[1] | 0 | REVERT_CODE_PREFIX | 0 | revert_code
-    // Start prefix is "tx_start".to_field() => 8 bytes
-    let prefix_bytes = TX_START_PREFIX.to_be_bytes::<8>();
-    let length_bytes = (offset as Field).to_be_bytes::<2>();
+    let expected_revert_code =
+        generate_padded_revert_code(offset as Field, tx_effect.revert_code as Field);
     // REVERT CODE
-    assert_eq(
-        tx_effects_hash_input[0],
-        field_from_bytes(
-            array_concat(
-                prefix_bytes,
-                [
-                    0,
-                    length_bytes[0],
-                    length_bytes[1],
-                    0,
-                    REVERT_CODE_PREFIX,
-                    0,
-                    tx_effect.revert_code,
-                ],
-            ),
-            true,
-        ),
-    );
+    assert_eq(tx_effects_hash_input[0], expected_revert_code);
 
     (tx_effects_hash_input, offset)
 }
 
+fn generate_padded_revert_code(offset: Field, revert_code: Field) -> Field {
+    // TX_START_PREFIX | 0 | 0 | 0 | 0 | REVERT_CODE_PREFIX | 0 | 0
+    let constant = (TX_START_PREFIX as Field) * (256 * 256 * 256 * 256 * 256 * 256 * 256)
+        + (REVERT_CODE_PREFIX as Field) * (256 * 256);
+
+    let padded_revert_code = constant + offset * (256 * 256 * 256 * 256) + revert_code;
+
+    padded_revert_code
+}
+
 unconstrained fn get_tx_effects_hash_input_helper(
-    tx_effect: TxEffect,
+    tx_hash: Field,
+    prefixed_tx_fee: Field,
+    note_hashes: [Field; MAX_NOTE_HASHES_PER_TX],
+    nullifiers: [Field; MAX_NULLIFIERS_PER_TX],
+    l2_to_l1_msgs: [Field; MAX_L2_TO_L1_MSGS_PER_TX],
+    public_data_update_requests: [PublicDataWrite; MAX_TOTAL_PUBLIC_DATA_UPDATE_REQUESTS_PER_TX],
+    private_logs: [Log<18>; MAX_PRIVATE_LOGS_PER_TX],
```

Comment:
I think this be `PRIVATE_LOG_SIZE_IN_FIELDS` rather than 18 (if we update the constant, this will Big Break)

Reply:
Ah I missed this one. Thanks.